src/httpclient.cpp

```diff
@@ -0,0 +1,252 @@
+// Copyright 2023 The Forgotten Server Authors.All rights reserved.
+// Use of this source code is governed by the GPL-2.0 License that can be found
+// in the LICENSE file.
+
+#include "otpch.h"
+
+#include "httpclient.h"
+
+#include "httpclientlib.h"
+
+#include "tasks.h"
+extern Dispatcher g_dispatcher;
+
+void HttpClient::threadMain()
+{
+	HttpClientLib::Request requestsHandler(
+	    [this](const HttpClientLib::HttpResponse_ptr& response) { clientRequestSuccessCallback(response); },
+	    [this](const HttpClientLib::HttpResponse_ptr& response) { clientRequestFailureCallback(response); });
+
+	std::unique_lock<std::mutex> requestLockUnique(requestLock, std::defer_lock);
+	while (getState() != THREAD_STATE_TERMINATED) {
+		requestLockUnique.lock();
+
+		if (pendingRequests.empty() && pendingResponses.empty()) {
+			requestSignal.wait(requestLockUnique);
+		}
+
+		if (!pendingRequests.empty() || !pendingResponses.empty()) {
```

Comment:
Early return unlock

Reply:
I didn`t get on how the early return would be placed here. Could you please add an example?